### PR TITLE
Implement Priors from "Vanilla Bayesian Optimization Performs Great in High Dimensions"

### PR DESCRIPTION
**Pull request by jduerholt**
This PR implements the dimensionality scaled priors from the paper  "Vanilla Bayesian Optimization Performs Great in High Dimensions" from Hvarfner et al. (https://arxiv.org/pdf/2402.02229) This is also a nice blogpost regarding the method:  https://www.miguelgondu.com/blogposts/2024-03-16/when-does-vanilla-gpr-fail/

Probably, I will set the prior also as new default.

### CODE REVIEW

**Comment by bertiqwerty** [comment]
That's a great idea!

**Comment by jduerholt** [comment]
Anything missing fore merging it in? Or do you need a bit more time for review?

File: bofire/data_models/priors/api.py [hunk 1]
```diff
@@ -12,8 +16,14 @@
     GammaPrior,
     NormalPrior,
     LKJPrior,
+    LogNormalPrior,
+    DimensionalityScaledLogNormalPrior,
 ]
 
+# these are priors that are generally applicable
+# and does not depend on problem specific extrea parameters
```
Comment (bertiqwerty) [inline bofire/data_models/priors/api.py:24]: ...and do not depend...

**Review by bertiqwerty — APPROVED** [review]
Looks good!

File: bofire/data_models/priors/api.py (hunk 1 above)
Comment (jduerholt) [reply bofire/data_models/priors/api.py]: Fixed it. Thanks!

**Pull request merged**